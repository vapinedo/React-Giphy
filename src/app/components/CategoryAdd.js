import React, { useState } from 'react'

export const CategoryAdd = () => {

    const [inputValue, setInputValue] = useState("Hello world");

    const handelInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(e)
    };

    return (
        <form onSubmit={ handleSubmit } className="my-4">
            <input 
                type="text" 
                value={ inputValue }
                className="form-control" 
                onChange={ handelInputChange }
                placeholder="Search gif..." />
        </form>
    )
}
