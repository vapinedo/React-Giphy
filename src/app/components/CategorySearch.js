import React, { useState } from 'react'

export const CategorySearch = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("");

    const onSearching = (e) => {
        const query = e.target.value;
        setInputValue(query);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length < 2) return;
        setCategories(prevCategories => [...prevCategories, inputValue]);
        setInputValue("");
    };

    return (
        <form onSubmit={ onSubmit } className="my-4">
            <input 
                type="text" 
                value={ inputValue }
                className="form-control" 
                onChange={ onSearching }
                placeholder="Search gif ..." />
        </form>
    )
}
