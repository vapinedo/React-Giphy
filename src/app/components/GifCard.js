import React from 'react'
import PropTypes from "prop-types";

export const GifCard = ({ title, url }) => {
    return (
        <article className="col-sm-6 col-md-4 col-lg-3 animate__animated animate__fadeIn animate_slow">
            <div className="card h-100">
                <img src={ url } className="card-img-top" alt={ title } />

                <div className="card-body">
                    <h5 className="card-title">{ title }</h5>
                </div>
            </div>
        </article>
    )
}

GifCard.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}