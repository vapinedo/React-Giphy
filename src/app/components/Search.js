import React, { useState } from 'react'
import PropTypes from "prop-types";

export const Search = ({ setCategory }) => {

    const [inputValue, setInputValue] = useState("");

    const onSearching = (e) => {
        const query = e.target.value;
        setInputValue(query);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length < 2) return;
        setCategory(inputValue);
        setInputValue("");
    };

    return (
        <section className="row">
            <form onSubmit={ onSubmit } className="col-md-12 my-4">
                <input 
                    type="text" 
                    value={ inputValue }
                    className="form-control" 
                    onChange={ onSearching }
                    placeholder="Search gif ..." />
            </form>
        </section>
    )
}

Search.propTypes = {
    setCategory: PropTypes.func.isRequired
};  