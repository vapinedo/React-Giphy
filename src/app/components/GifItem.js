import React, { useState, useEffect } from 'react'
import { GifCard } from '@components/GifCard';

export const GifItem = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGif();
    }, [])

    const searchEndPoint = "gifs/search";
    const baseUrl = "https://api.giphy.com/v1/";
    const apiKey = "KMuvgZhmZXBQ8wLRPBeZPcVDCwSJPfcc";

    const getGif = async(category) => {
        const request = await fetch(`
            ${baseUrl}${searchEndPoint}?q=Rick+and+Morty&limit=10&api_key=${apiKey}`
        );
        const { data } = await request.json();
        const gifs = data.map(gif => {
            return {
                id: gif.id,
                title: gif.title,
                url: gif.images?.downsized_medium.url
            }
        });
        console.log(gifs);
        setImages(gifs);
    };

    return (
        <section className="row g-4">
            {images.map(img => (
                <GifCard key={ img.id } { ...img } />
            ))}
        </section>
    )
}
