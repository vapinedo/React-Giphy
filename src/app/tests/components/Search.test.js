import React from "react";
import { shallow } from "enzyme";
import { Search } from "../../components/Search";

describe("Unit testing for <Search /> component", () => {

    const setCategory = () => {};

    test("Should render correctly", () => {
        const wrapper = shallow(<Search setCategory={setCategory} />);
        expect(wrapper).toMatchSnapshot();
    });
});