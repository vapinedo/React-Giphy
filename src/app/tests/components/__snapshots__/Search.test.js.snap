// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing for <Search /> component Should render correctly 1`] = `
<section
  className="row"
>
  <form
    className="col-md-12 my-4"
    onSubmit={[Function]}
  >
    <input
      className="form-control animate__animated animate__fadeIn"
      onChange={[Function]}
      placeholder="Search gif ..."
      type="text"
      value=""
    />
  </form>
</section>
`;
